// @hashgraph/sdk/lib/client/WebClient.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Network = void 0;
var _Client = _interopRequireDefault(require("./Client.cjs"));
var _WebChannel = _interopRequireDefault(require("../channel/WebChannel.cjs"));
var _AccountId = _interopRequireDefault(require("../account/AccountId.cjs"));
var _LedgerId = _interopRequireDefault(require("../LedgerId.cjs"));
var _ClientConstants = require("../constants/ClientConstants.cjs");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
// SPDX-License-Identifier: Apache-2.0

// eslint-disable-next-line @typescript-eslint/no-unused-vars

/**
 * @typedef {import("./Client.js").ClientConfiguration} ClientConfiguration
 */

const Network = exports.Network = {
  /**
   * @param {string} name
   * @returns {{[key: string]: (string | AccountId)}}
   */
  fromName(name) {
    switch (name) {
      case "mainnet":
        return Network.MAINNET;
      case "testnet":
        return Network.TESTNET;
      case "previewnet":
        return Network.PREVIEWNET;
      default:
        throw new Error(`unknown network name: ${name}`);
    }
  },
  MAINNET: _ClientConstants.MAINNET,
  TESTNET: _ClientConstants.WEB_TESTNET,
  PREVIEWNET: _ClientConstants.WEB_PREVIEWNET
};

/**
 * Represents a client for interacting with the Hedera network over the web.
 * The `WebClient` class extends the base `Client` class and provides methods
 * for configuring and managing connections to the Hedera network, including
 * setting the network type (mainnet, testnet, previewnet) and handling
 * transactions and queries.
 * @augments {Client<WebChannel, *>}
 */
class WebClient extends _Client.default {
  /**
   * @param {ClientConfiguration} [props]
   */
  constructor(props) {
    super(props);
    if (props != null) {
      if (typeof props.network === "string") {
        switch (props.network) {
          case "mainnet":
            this.setNetwork(Network.MAINNET);
            this.setLedgerId(_LedgerId.default.MAINNET);
            break;
          case "testnet":
            this.setNetwork(Network.TESTNET);
            this.setLedgerId(_LedgerId.default.TESTNET);
            break;
          case "previewnet":
            this.setNetwork(Network.PREVIEWNET);
            this.setLedgerId(_LedgerId.default.PREVIEWNET);
            break;
          default:
            throw new Error(
            // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
            `unknown network: ${props.network}`);
        }
      } else if (props.network != null) {
        this.setNetwork(props.network);
      }
    }
  }

  /**
   * @param {string | ClientConfiguration} data
   * @returns {WebClient}
   */
  static fromConfig(data) {
    return new WebClient(typeof data === "string" ? (/** @type {ClientConfiguration | undefined} */
    JSON.parse(data)) : data);
  }

  /**
   * Construct a client for a specific network.
   *
   * It is the responsibility of the caller to ensure that all nodes in the map are part of the
   * same Hedera network. Failure to do so will result in undefined behavior.
   *
   * The client will load balance all requests to Hedera using a simple round-robin scheme to
   * chose nodes to send transactions to. For one transaction, at most 1/3 of the nodes will be
   * tried.
   *
   * @param {{[key: string]: (string | AccountId)} | string} network
   * @returns {WebClient}
   */
  static forNetwork(network) {
    return new WebClient({
      network,
      scheduleNetworkUpdate: false
    });
  }

  /**
   * @param {string} network
   * @returns {WebClient}
   */
  static forName(network) {
    return new WebClient({
      network,
      scheduleNetworkUpdate: false
    });
  }

  /**
   * Construct a Hedera client pre-configured for Mainnet access.
   *
   * @returns {WebClient}
   */
  static forMainnet() {
    return new WebClient({
      network: "mainnet",
      scheduleNetworkUpdate: false
    });
  }

  /**
   * Construct a Hedera client pre-configured for Testnet access.
   *
   * @returns {WebClient}
   */
  static forTestnet() {
    return new WebClient({
      network: "testnet",
      scheduleNetworkUpdate: false
    });
  }

  /**
   * Construct a Hedera client pre-configured for Previewnet access.
   *
   * @returns {WebClient}
   */
  static forPreviewnet() {
    return new WebClient({
      network: "previewnet",
      scheduleNetworkUpdate: false
    });
  }

  /**
   * @param {{[key: string]: (string | AccountId)} | string} network
   * @returns {void}
   */
  setNetwork(network) {
    if (typeof network === "string") {
      switch (network) {
        case "previewnet":
          this._network.setNetwork(Network.PREVIEWNET);
          break;
        case "testnet":
          this._network.setNetwork(Network.TESTNET);
          break;
        case "mainnet":
          this._network.setNetwork(Network.MAINNET);
      }
    } else {
      this._network.setNetwork(network);
    }
  }

  /**
   * @param {string[] | string} mirrorNetwork
   * @returns {this}
   */
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  setMirrorNetwork(mirrorNetwork) {
    if (typeof mirrorNetwork === "string") {
      this._mirrorNetwork.setNetwork([]);
    } else {
      this._mirrorNetwork.setNetwork(mirrorNetwork);
    }
    return this;
  }

  /**
   * @override
   * @returns {(address: string) => WebChannel}
   */
  _createNetworkChannel() {
    return address => new _WebChannel.default(address);
  }

  /**
   * @override
   * @returns {(address: string) => *}
   */
  _createMirrorNetworkChannel() {
    return () => {
      throw new Error("mirror support is not supported in browsers");
    };
  }

  /**
   * @override
   * @returns {Promise<void>}
   */
  updateNetwork() {
    return Promise.reject(new Error("Update network is not supported in browsers"));
  }
}
exports.default = WebClient;
